remme/tp/pub_key.py

```diff
@@ -318,6 +330,91 @@ def _store_pub_key(self, context, signer_pubkey, transaction_payload):
 
         return state
 
+    def _store_public_key_for_other(self, context, signer_pubkey, transaction_payload):
```

Comment:
This thing duplicates the whole logic of `_store_pub_key`. Please reduce duplication.